### PR TITLE
[api] Provisioning templates test refactor

### PR DESCRIPTION
**Pull request by rplevka**
- replacing deprecated `config_template` references by `provisioning_templates`
- refactor crud tests for api `ProvisioningTemplate` tests
- unittest to pytest conversion for api `ProvisioningTemplate` tests
- refactor `test_positive_build_pxe_default` test case

```
Testing started at 12:10 AM ...
/home/rplevka/.virtualenvs/robottelo/bin/python /usr/share/java/pycharm-community/plugins/python-ce/helpers/pycharm/_jb_pytest_runner.py --target test_template.py::TestProvisioningTemplate
Launching pytest with arguments test_template.py::TestProvisioningTemplate in /home/rplevka/work/rplevka/robottelo/tests/foreman/api

============================= test session starts ==============================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1 -- /home/rplevka/.virtualenvs/robottelo/bin/python
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/rplevka/work/rplevka/robottelo/tests/foreman, inifile: pytest.ini
plugins: xdist-1.31.0, services-1.3.1, repeat-0.8.0, mock-1.10.4, cov-2.8.1, forked-1.1.3
collecting ... 2020-05-18 22:10:54 - conftest - DEBUG - Collected 2 test cases
...
collected 2 items

test_template.py::TestProvisioningTemplate::test_positive_end_to_end_crud 
test_template.py::TestProvisioningTemplate::test_positive_build_pxe_default 

========================== 2 passed in 66.11 seconds ===========================
```

### CODE REVIEW

File: tests/foreman/api/test_template.py [hunk 1]
```diff
@@ -42,303 +44,232 @@
...
         ).create()
-        # Update template name with that user
-        cfg = get_nailgun_config()
-        cfg.auth = (user_login, user_password)
+        assert template.name == name
+        assert len(template.organization) == 1, "Template should be assigned to a single org here"
+        assert template.organization[0].id == module_org.id
+        assert len(template.location) == 1, "Template should be assigned to a single location here"
+        assert template.location[0].id == module_location.id
+        assert template.snippet is False, "Template snippet attribute is True instead of False"
```
Comment (mirekdlugosz) [inline tests/foreman/api/test_template.py:195]:
```python
assert not template.snippet
```
?
Or `assert template.snippet == False`. Linter might complain about this one. But seems more natural, as we want to compare values, not if specific variable points to same object.

File: tests/foreman/api/test_template.py [hunk 2]
```diff
@@ -42,303 +44,232 @@
...
+
+    :param module_org:
+    :return: A dictionary containing nailgun entities, names, values and paths of the custom
+    Templates
+    """
+    tftpboot_path = '/var/lib/tftpboot'
+    default_templates = {
+        'pxegrub': {
+            'setting': 'global_PXEGrub',
+            'path': '{0}/grub/menu.lst'.format(tftpboot_path),
```
Comment (mirekdlugosz) [inline tests/foreman/api/test_template.py:103]: These look like good candidate for f-string.

File: tests/foreman/api/test_template.py [hunk 3]
```diff
@@ -42,303 +44,232 @@
...
-            not updated.
-
-        :CaseImportance: Medium
+        assert updated.name == new_name, "The Provisioning template wasn't properly renamed"
+        # clone
+        template_origin = template.read_json()
+        # remove unique keys
+        unique_keys = ('updated_at', 'created_at', 'id', 'name')
+        for key in unique_keys:
+            del template_origin[key]
```
Comment (mirekdlugosz) [inline tests/foreman/api/test_template.py:222]: dict comprehension?
```python
template_origin = template.read_json()
unique_keys = ('updated_at', 'created_at', 'id', 'name')
template_origin = {key: value for key, value in template_origin.items() if key not in unique_keys}
```
Dunno, feels more pythonic.

File: tests/foreman/api/test_template.py (hunk 1 above)
Comment (rplevka) [reply tests/foreman/api/test_template.py:195]: If i recall correctly, my version was suggested by the linter :)

File: tests/foreman/api/test_template.py (hunk 3 above)
Comment (rplevka) [reply tests/foreman/api/test_template.py:222]: you are right, this was the original code structure and i didn't update it. I will do it now

File: tests/foreman/api/test_template.py (hunk 1 above)
Comment (mirekdlugosz) [reply tests/foreman/api/test_template.py:195]: Oh, ok. It's not a huge problem anyway.

**Pull request merged**